### ISSUE TITLE
New Viewer MVP

### ISSUE THREAD

**Issue by knowtheory** [documentcloud/documentcloud#239]
This is a discussion for getting to a minimalist viewer replacement as quickly as possible.  Our initial goal is to smash out a priority list of requirements based on the work we've done over the past year to improve individual frontend components of the platform (note embeds and page embeds in particular).

At this point, DocumentCloud's embeddable notes are responsive and react to viewport width appropriately (encouraging more folks to make notes is a separate issue).  Over the past couple of months, we've also worked on a single page embed which is also responsive and displays notes correctly.

Using that work as a starting point, what should a minimum viable product replacement for the current viewer look like?

Here is an initial proposal:

1. [ ] Displays page images
2. [ ] pinch to zoom on mobile devices
3. [ ] displays notes as touchable/clickable regions which fold open w/ annotation text
4. [ ] navigate between pages
5. [ ] search document text

**Comment by anthonydb** [comment]
Two more possibilities:
1 [] Link to download or view full document
2 [] DocumentCloud logo

**Comment by reefdog** [comment]
So to be clear, these are the requirements for launching a viewer which will eventually replace the existing viewer, but which at launch need not have all the existing viewer's features, because the existing viewer will still be usable. Right?

I'd actually remove searching from MVP. It's necessary and should come soon after, but I'd be okay launching without it. I'd add:

- [ ] Responsive (implied, but should be explicitly stated)
- [ ] View page text

**Comment by knowtheory** [comment]
Yep that's right, i'll add those to the list.

**Comment by anthonydb** [comment]
Yes on the product trajectory of the viewer -- eventual replacement of the current viewer but existing alongside as an outgrowth of page embed to start.

**Comment by reefdog** [comment]
+ iframable

**Comment by knowtheory** [comment]
Okay, where in the priority list does iframeable go?  IMO it's down lower on the list, since it's mostly about our ability to control styling and not interfere with existing page.  But since our viewer currently lives directly in DOM already, switching it doesn't seem super high priority to me.

That said, if its structurally important to the way we wish to go about our strategy of responsiveness, we can bump it up the list.

**Comment by reefdog** [comment]
The reason I'd put it mid-to-high on the list is that I don't want it to fall into the "nice to have" pile and never get done. It's apparently critical to several use cases.

**Comment by knowtheory** [comment]
Yep, everything on this list needs to be done.  the question is order, and what makes the cut before the first deployment.
